### DISCUSSION TITLE
Expect to find '=' token

### DISCUSSION THREAD

**Discussion by saigkill** [cake-build/cake#3883] [Q&A] [answered]
Hello,
i just refactored my app to match a new name schema. Currently i'm using that build.cake:

`///////////////////////////////////////////////////////////////////////////////
// ARGUMENTS
///////////////////////////////////////////////////////////////////////////////

var target = Argument("target", "Default");
var configuration = Argument("configuration", "Release");
var targetframework = Argument("targetframework", "net6.0-windows10.0.20348.0");

///////////////////////////////////////////////////////////////////////////////
// SETUP / TEARDOWN
///////////////////////////////////////////////////////////////////////////////

Setup(ctx =>
{
	// Executed BEFORE the first task.
	Information("Running tasks...");
});

Teardown(ctx =>
{
	// Executed AFTER the last task.
	Information("Finished running tasks.");
});

///////////////////////////////////////////////////////////////////////////////
// TASKS
///////////////////////////////////////////////////////////////////////////////

Task("Clean")	
	.Does(() =>
	{
		CleanDirectory($"./bin/publish/");		
	});

Task("Build")
	.IsDependentOn("Clean")
	.Does(() =>
	{
		DotNetCoreBuild("../WayfairShippingApp.sln", new DotNetCoreBuildSettings
		{
			Configuration = configuration,
		});
	});

Task("Publish")
	.IsDependentOn("Build")
	.Does(() =>
	{
		DotNetCorePublish(".", new DotNetCorePublishSettings
		{
			Framework = "net6.0-windows10.0.20348.0",
			Configuration = "Release",
			OutputDirectory = "./bin/publish",
			Runtime = "win10-x86",
			SelfContained = false
		});
	});

Task("Publish64")
	.IsDependentOn("Build")
	.Does(() =>
	{
		DotNetCorePublish(".", new DotNetCorePublishSettings
		{
			Framework = "net6.0-windows10.0.20348.0",
			Configuration = "Release",
			OutputDirectory = "./bin/publish64",
			Runtime = "win10-x64",
			SelfContained = false
		});
	});

Task("Default")
	.IsDependentOn("Clean")
	.IsDependentOn("Build")
	.IsDependentOn("Publish")	
	.IsDependentOn("Publish64")
	.Does(() => 
	{
	});

RunTarget(target);
`
Currently a dotnet cake results in a Error: Expected to find '=' token. Maybe i missed anything?

**Comment by devlead** [comment, marked as answer]
The script compiles for me, at what point are you seeing the error? Is it before setup / any tasks are executed? If so how are you calling your script?
I took your script and executed the following 
```bash
dotnet cake --target=Clean
```
and got the following output
```
----------------------------------------
Setup
----------------------------------------
Running tasks...

========================================
Clean
========================================

----------------------------------------
Teardown
----------------------------------------
Finished running tasks.

Task                          Duration
--------------------------------------------------
Setup                         00:00:00.0165123
Clean                         00:00:00.0111020
Teardown                      00:00:00.0003029
--------------------------------------------------
Total:                        00:00:00.0279172
```

**Reply by saigkill** [reply]
In general it comes directly after the dotnet cake command:

`dotnet cake --target=Clean
Error: Expected to find '=' token.`

It already comes with --target=Clean.

**Reply by saigkill** [reply]
I found the error. We are using a automatic licenseheader extension. That extension thought cake.config is a XML file and placed a  Header there. After deleting that it works as expected:

`  dotnet cake --target=Clean

----------------------------------------
Setup
----------------------------------------
Running tasks...

========================================
Clean
========================================

----------------------------------------
Teardown
----------------------------------------
Finished running tasks.

Task                          Duration
--------------------------------------------------
Setup                         00:00:00.0060793
Clean                         00:00:00.0055260
Teardown                      00:00:00.0015809
--------------------------------------------------
Total:                        00:00:00.0131862`

Sorry for that trouble.

**Reply by devlead** [reply]
No worries, great that you got it sorted.
